openmdao/core/notebook_utils.py

```diff
@@ -0,0 +1,130 @@
+"""Checking for interactive notebook mode."""
+import sys
+import importlib
+import inspect
+
+from IPython.display import display, HTML, Code
+
+from openmdao.utils.general_utils import simple_warning
+
+try:
+    from tabulate import tabulate
+except ImportError:
+    tabulate = None
+
+colab = 'google.colab' in sys.modules
+
+
+def _get_object_from_reference(reference):
+    """
+    Return object of given reference path.
+
+    Parameters
+    ----------
+    reference : str
+        Dot path of desired class.
+
+    Returns
+    -------
+    str
+        Object of the given class.
+    """
+    split = reference.split('.')
+    right = []
+    module = None
+    while split:
+        try:
+            module = importlib.import_module('.'.join(split))
+            break
+        except ModuleNotFoundError:
+            right.append(split.pop())
+    if module:
+        for entry in reversed(right):
+            module = getattr(module, entry)
```

Comment:
Naming is a little confusing here.  You're not returning a module, but an object in a module, correct?